Assignment6.Rmd

```diff
@@ -70,16 +70,45 @@ table(M2$certified, M2$predict1)
 
 table(M2$certified, M2$predict2)
 
+sum(diag(table(M2$certified, M2$predict1)))/sum(table(M2$certified, M2$predict1))
+sum(diag(table(M2$certified, M2$predict2)))/sum(table(M2$certified, M2$predict2))
 ```
+The accuracy for the first model is 21.9% and for second model is 53.6%, therefore model 2 has a lower error rate.
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
-
+data <- read.csv("student.record.csv", header = T)
+data <- data[,c(4:13)]
+data<- na.omit(data)
+
+# for the first model, I use SAT scores, ACT scores and sex to predict GPA
+c.tree3 <- rpart(as.factor(HSGPA) ~., method = "class", data = data)
+printcp(c.tree3)
+
+# for the second model, I use composite SAT scores, composite ACT scores and sex to predict GPA
+HSGPA <- data[,1]
+SEX <- data[, 10]
+ACT <- data[, 2:6]
+ACT$ACTscore <- rowSums(ACT)
+SAT <- data[, 7:9]
+SAT$SATscore <- rowSums(SAT)
+data2 <- cbind(HSGPA, ACT, SAT, SEX)
+c.tree4 <- rpart(as.factor(HSGPA) ~ ACTscore + SATscore + SEX, 
+                 method = "class", data = data2)
+printcp(c.tree4)
+
+data2$predict1 <- predict(c.tree3, data2, type = "class")
+t1 <- table(data2$HSGPA, data2$predict1)
+sum(diag(table(data2$HSGPA, data2$predict1)))/sum(table(data2$HSGPA, data2$predict1))
+
+data2$predict2 <- predict(c.tree4, data2, type = "class")
+t2 <- table(data2$HSGPA, data2$predict2)
+sum(diag(table(data2$HSGPA, data2$predict2)))/sum(table(data2$HSGPA, data2$predict2))
 ```
-
+The accuracy of the first model is 26.08%. The accuracy of the second model is 26.04%. Two models are having almost the same accuracy, but considering the parsimony of the model, second model is better because it uses fewer variables to predict the outcome.
```

Comment:
Good work. It seems you have a good understanding of model comparison. It is good that you pay attention to model parsimony when the accuracy rates are similar. Both of your models have a cv error rate of more than 0.95, which is very high. Could you think of any way that can help to improve this?